sql/pgaudit.sql

```diff
@@ -818,6 +818,68 @@ RESET ROLE;
 DROP TABLE public.t;
 DROP ROLE alice;
 
+-- Test PARTITIONED table
+CREATE TABLE h(x int ,y int) PARTITION BY HASH(x);
+CREATE TABLE h_0 partition OF h FOR VALUES WITH ( MODULUS 2, REMAINDER 0); 
+CREATE TABLE h_1 partition OF h FOR VALUES WITH ( MODULUS 2, REMAINDER 1); 
+INSERT INTO h VALUES(1,1);
+SELECT * FROM h;
+SELECT * FROM h_0;
+CREATE INDEX h_idx ON h (x);
+CREATE INDEX h_0_idx_y on h_0 (y);
+DROP INDEX h_0_idx_y;
+DROP INDEX h_idx;
+DROP TABLE h;
+
+SET pgaudit.log = 'ALL';
+create table h(a int ,b int, c int,d int, e int, f int) partition by hash(a);
+create table h_0 partition of h for values with ( modulus 2, remainder 0) partition by hash(b); 
+create table h_0_0 partition of h_0 for values with(modulus 2, remainder 0) partition by hash(c);
+create table h_0_0_0 partition of h_0_0 for values with (modulus 2, remainder 0);
+create table h_0_0_1 partition of h_0_0 for values with (modulus 2, remainder 1);
+create table h_0_1 partition of h_0 for values with(modulus 2, remainder 1);
+create table h_1 partition of h for values with ( modulus 2, remainder 1) partition by hash(d);
+create table h_1_0 partition of h_1 for values with(modulus 2, remainder 0);
+create table h_1_1 partition of h_1 for values with(modulus 2, remainder 1) partition by hash(e);
+create table h_1_1_0 partition of h_1_1 for values with (modulus 2, remainder 0);
+create table h_1_1_1 partition of h_1_1 for values with (modulus 2, remainder 1) partition by hash(f);
+create table h_1_1_1_0 partition of h_1_1_1 for values with (modulus 2, remainder 0);
+create table h_1_1_1_1 partition of h_1_1_1 for values with (modulus 2, remainder 1);
+select * from h;
+select * from h_0;
+select * from h_1_1_1_1;
+create table g(x int);
+select * from g;
+select * from h,g;
+select * from h_0,g;
+select * from h_0_1,g;
+select * from g,h_1_1_1;
+select * from g,h_1_1_1_0;
+drop table g;
+drop table h;
+
+-- Test inherited table
+CREATE TABLE a(aid  int,aname varchar(20));
+CREATE TABLE b(bid int)INHERITS(a);
+insert into a values(0,'Dava');
+insert into b values(1,'April',100);
+select * from a;
+select * from b;
+
+CREATE TABLE g(x int ,y int) PARTITION BY HASH(x);
+CREATE TABLE g_0 partition OF g FOR VALUES WITH ( MODULUS 2, REMAINDER 0);
```

Comment:
Why more partition tests here? What's different from above?

Reply:
this case is for inherited table, and also add the interaction with partitioned table.  
Test 'JOIN' to make sure the audit is correct when join inherited table with partitioned table.